### ISSUE TITLE
Buf doesn't seem to support use case where your module uses multiple .proto files

### ISSUE THREAD

**Issue by ahctangU** [bufbuild/buf#507]
Essentially, I have very large .proto files that are split into 4 files to keep the developers sane, like so:

With protoc I can set -I to my api1/v1 folder and stop worrying about it. Previously with buf, I could also hack it with the `build.roots`, but how is this supposed to work now?

`service` imports both requests.proto and responses.proto.

This use case doesn't seem to be supported, but it seems like it would be something someone would do...
```
.
├── api1
│   ├── v1
│   │   └── service.proto
|   |  └── requests.proto
|   |  └── responses.proto
│   └── buf.yaml
```

**Comment by bufdev** [comment]
Of course buf supports multiple files. What is the actual bug? Please file all issues with reproducible examples.

Closing until there's a specific, reproducible example. Happy to re-open if one is added.

**Comment by bufdev** [comment]
Roots have been replaced with workspaces https://docs.buf.build/configuration/v1beta1-migration-guide#buildroots

However, we havent removed `v1beta1`, so you could just keep on using `v1beta1` `buf.yaml` files.

**Comment by agenslerDNA** [comment]
I have this issue too whenever multiple proto files are involved. I'm hoping that it's a how to. I didn't have an issue with the previous releases, but with this new one, it's tough to figure out how this is supposed to work.

There are a few scenarios here, A.proto imports B, C, and D.
- topdir
  - subdir
    - v1
      - A.proto
      - B.proto
      - C.proto
      - D.proto
- buf.yaml (is in the topdir)

buf.yaml looks like this
`version: v1
breaking:
  use:
    - FILE
lint:
  use:
    - DEFAULT`

buf lint will give an error of "subdir/v1/A.proto:7:8:B.proto: does not exist"
if I add a buf.work.yaml file to topdir, I see these errors with buf lint

Files with package "subdir.v1" must be within a directory "subdir/v1" relative to root but were in directory ".".

if I add a buf.yaml file in the subdir/v1 directory
`version: v1
breaking:
  use:
    - FILE
lint:
  use:
    - DEFAULT
  except:
    - PACKAGE_DIRECTORY_MATCH`

buf lint will work, but now I'm not checking for something I definitely want to check for. The proto files are actually in the correct place at the end of the day. I am however; able to do a break check against an image.

The other scenario related to multiple proto files is when the proto files are in different subdirectories (import is involved here too). Follow the same directions as above, but in this case protoA is importing protoB in a different subdirectory:

- top
  - sub1
    - v1
      - protoA.proto
      - buf.yaml
  - sub2
      - v1
        - protoB.proto
        - buf.yaml
- buf.work.yaml

buf.work.yaml looks like the below. It's this example is simplified, there are more directories between topdir and sub# on my end.

`version: v1
directories:
  - sub1/v1
  - sub2/v1`

When running a buf break check against the image, I'm seeing this error:
"Failure: input contained 2 images, whereas against contained 1 images"

The image was created with `buf build -o something.bin.gz`, and the check is done with `buf breaking --against something.bin.gz`

**Comment by amckinney** [comment]
For your first problem, imports are relative to each `buf.yaml` root. So if your `buf.yaml` is defined in `topdir`, you would need to import `B.proto` as `subdir/v1/B.proto`.

As for your second problem, the error you're seeing is because we currently require there to be the same number of workspace directories in both the target input and the `--against` input. If you're adding a new directory to your `buf.work.yaml`, you won't be able to compare it to it's previous state (since it has a different number of images). In its current state, images are individually compared against each other to ensure compatibility.

We've discussed lifting the "Failure: input contained 2 images, whereas against contained 1 images" constraint in the past, so this is some extra motivation. I created a [new issue](https://github.com/bufbuild/buf/issues/567) to track this - we'll look into it soon!
